### DISCUSSION TITLE
Using existing Godot demos for testing

### DISCUSSION THREAD

**Discussion by Saitodepaula** [godot-jolt/godot-jolt#169] [General]
Are there any chance that you could use some existing demos for Godot to test Godot-Jolt?

May I suggest my own [demo](https://github.com/Saitodepaula/Godot-6DOF-Vehicle-Demo)? It's a physics based vehicle using only 6DOF joints. 

In my demo's github page, I also list other vehicle implementations (using other methods) and issues with Bullet and Godot Physics. So it could be a nice source of material for testing.

I know it sounds like I'm lazy and just want you to do the hard work for me, but since my demo has more than 50 stars on github, I think there is some interest in it.

Also, since my demo uses many of the 6DOF features, it can be a good test I think. 

Another point: the vehicles have different behaviour with different physics FPS, so it could be yet another type of test to the physics engine.

**Comment by mihe** [comment]
Yes, the existing examples are a bit lackluster. The long-term plan was to port over Jolt's [own samples](https://github.com/jrouwe/JoltPhysics/blob/d5cf4634292ad9a35421ffceb42a50faf3ce5562/Docs/Samples.md) to the examples project, mostly as a sanity check. These include a constraint-based vehicle, similar to your own I guess, as well as some ray-cast vehicles. The ray-cast vehicles do however use a specific `VehicleConstraint` in Jolt that probably won't be exposed in the first phase of this project, but Godot's own `VehicleBody3D` might suffice there.

As far as including existing Godot demos, I've also been contemplating making use of the [official physics tests](https://github.com/godotengine/godot-demo-projects/tree/4.0-dev/3d/physics_tests) in some way, and maybe some other demo project there. I guess the "Truck Town" demo would be an interesting test for the `VehicleBody3D` node type, even if it's a bit rough around the edges.

All that is to say that improving the examples/demos is on my list of things to do, as well adding some kind of functional tests, but probably for when the extension is a bit more stable. And while I might not use your specific demo directly, there will most likely be some kind of constraint vehicle and ray-cast vehicle in there.
